packages/x-date-pickers/src/internals/hooks/useClearEndAdornment/useClearEndAdornment.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+
+import { useSlotProps } from '@mui/base';
+import ClearIcon from '@mui/icons-material/Clear';
+import IconButton from '@mui/material/IconButton';
+import { SlotsAndSlotProps } from '../../utils/slots-migration';
+import { FieldSlotsComponents, FieldSlotsComponentsProps } from '../useField/useField.types';
+
+type UseClearEndAdornmentProps<
+  TInputProps extends { endAdornment?: React.ReactNode } | undefined,
+  TFieldSlotsComponents extends FieldSlotsComponents,
+  TFieldSlotsComponentsProps extends FieldSlotsComponentsProps,
+> = {
+  clearable: boolean;
+  InputProps: TInputProps;
+  onClear: React.MouseEventHandler<HTMLButtonElement>;
+} & SlotsAndSlotProps<TFieldSlotsComponents, TFieldSlotsComponentsProps>;
+
+export const useClearEndAdornment = <
+  TInputProps extends { endAdornment?: React.ReactNode } | undefined,
+  TFieldSlotsComponents extends FieldSlotsComponents,
+  TFieldSlotsComponentsProps extends FieldSlotsComponentsProps,
+>({
+  clearable,
+  InputProps: ForwardedInputProps,
+  onClear,
+  slots,
+  slotProps,
+}: UseClearEndAdornmentProps<TInputProps, TFieldSlotsComponents, TFieldSlotsComponentsProps>) => {
+  const EndClearIcon = slots?.clearIcon ?? ClearIcon;
+  const endClearIconProps = useSlotProps({
+    elementType: ClearIcon,
+    externalSlotProps: slotProps?.clearIcon,
+    externalForwardedProps: {},
+    ownerState: {},
+  });
+
+  const InputProps = {
+    ...ForwardedInputProps,
+    endAdornment: clearable ? (
+      <React.Fragment>
+        {ForwardedInputProps?.endAdornment}
+        <IconButton className="clearButton" onClick={onClear} tabIndex={-1}>
+          <EndClearIcon {...endClearIconProps} />
```

Comment:
The icon seems too big (bigger than on the Autocomplete) 🙈 
```suggestion
          <EndClearIcon size="small" {...endClearIconProps} />
```

Reply:
Makes sense, but do we want users to be able to override this? 
And wouldn't this create visual inconsistencies with other `endAdornments` that might be there? 🤔

Reply:
> Makes sense, but do we want users to be able to override this?

It would still be overrideable via the `slotProps`, that's why the `size` is before the other props spread.

> Makes sense, but do we want users to be able to override this? And wouldn't this create visual inconsistencies with other `endAdornments` that might be there? 🤔

I wouldn't say that it would look out of place... Would you? 🤔 
![Screenshot 2023-06-01 at 15 32 32](https://github.com/mui/mui-x/assets/4941090/b854998e-2ff3-4b7c-a097-81752b1decfe)

Reply:
In which case I would add it in the `additionalProps` or `const endClearIconProps = useSlotProps`